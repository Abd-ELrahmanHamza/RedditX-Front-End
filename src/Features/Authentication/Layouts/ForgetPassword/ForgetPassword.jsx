import { useState, useEffect } from "react";

import { SiReddit } from "react-icons/si";

import FormInputPageCom from "Features/Authentication/Components/FormInputPageCom/FormInputPageCom";
import Button from "Features/Authentication/Components/Button/Button";

import {
  AuthContainer,
  ButtonsContainer,
  ErrorParagraph,
  AuthParagraph,
  AuthHeader,
  Forget,
  RedditIcon,
} from "./ForgetPassword.styled";

const USER_REGEX = /^[A-z][A-z0-9-_]{3,20}$/;

const USER_EMAIL =
  /[a-zA-Z0-9._-]{3,}@[a-zA-Z0-9._-]{3,}[.]{1}[a-zA-Z0-9._-]{2,}/;

const defaultFormFields = {
  userName: "",
  password: "",
};

/**
 * ForgetPassword component that is used in forget password page
 * @returns {React.Component}  ForgetPassword component that is used in forget password page
 */

const ForgetPassword = () => {
  /**
   * state to know if the email is valid or not to control what to show to the user
   */
  const [validEmail, setValidEmail] = useState(false);

  /**
   * state to handel any change the user make in the input fields
   */

  const [formFields, setFormFields] = useState(defaultFormFields);

  /**
   * state to know if the userName is valid or not to control what to show to the user
   */

  const [validName, setValidName] = useState(false);

  /**
   * state to know if the user focused on input field at least one time
   */

  const [initialFocus, setInitialFocus] = useState(true);
  /**
   * state to know if the user focused on input field at least one time
   */

  const [initialFocus2, setInitialFocus2] = useState(true);
  
  const { userName, email } = formFields;

  /**
   * useEffect for email field to check if the email that the user entered is valid or not
   */

  useEffect(() => {
    setValidEmail(USER_EMAIL.test(email));
  }, [email]);

  /**
   * useEffect for userName field to check if the userName that the user entered is valid or not
   */

  useEffect(() => {
    setValidName(USER_REGEX.test(userName));
  }, [userName]);

  /**
   * Function to handle the submit of the form of forget password
   * @param {*} event
   */
  const handleSubmit = async (event) => {
    event.preventDefault();
  };

  /**
   * Function to handle any change on the input field of the forget password form (check if the userName or the email is valid or not)
   * @param {*} event
   */

  const handleChange = (event) => {
    const { name, value } = event.target;
    if (name === "userName") {
      setValidName(USER_REGEX.test(userName));
    }
    if (name === "email") {
      setValidEmail(USER_EMAIL.test(email));
    }
    setFormFields({ ...formFields, [name]: value });
  };

  return (
    <>
      <AuthContainer>
        <RedditIcon>
          <SiReddit size={60} color={"red"}></SiReddit>
        </RedditIcon>
        <AuthHeader>Reset your password</AuthHeader>
        <AuthParagraph>
          Tell us the username and email address associated with your Reddit
          account, and weâ€™ll send you an email with a link to reset your
          password.
        </AuthParagraph>

        <br></br>

        <form onSubmit={handleSubmit}>
          <FormInputPageCom
            valid={validName}
            initialFocus={initialFocus}
            showIcon={true}
            label="USERNAME"
            type="text"
            onChange={handleChange}
            name="userName"
            value={userName}
            onFocus={() => {
              setInitialFocus(false);
            }}
          />

          {/* Show error message if the userName is not valid and the user made a focus on the it's input field */}
          <ErrorParagraph valid={validName || initialFocus}>
            Username must be between 3 and 20 characters
          </ErrorParagraph>

          <FormInputPageCom
            label="EMAIL"
            valid={validEmail}
            initialFocus={initialFocus2}
            showIcon={true}
            type="email"
            onChange={handleChange}
            name="email"
            value={email}
            onFocus={() => {
              setInitialFocus2(false);
            }}
          />

          {/* Show error message if the email is not valid and the user made a focus on the it's input field */}

          <ErrorParagraph valid={validEmail || initialFocus2}>
            not a valid email
          </ErrorParagraph>

          <ButtonsContainer>
            {/** Show an enabled button if the email and userName are valid */}
            {validName && validEmail && (
              <Button
                page={true}
                valid={validName && validEmail}
                type="button"
                onClick={() => {}}
              >
                RESET PASSWORD
              </Button>
            )}

            {/** Show an disabled button if the email or userName are not valid */}

            {(!validName || !validEmail) && (
              <Button
                page={true}
                disabled
                valid={validName && validEmail}
                type="button"
              >
                RESET PASSWORD
              </Button>
            )}
          </ButtonsContainer>
          <br></br>
          <Forget>
            <span onClick={() => {}}>FORGOT USERNAME?</span>{" "}
          </Forget>
          <br></br>
          <br></br>
          <Forget>
            Don't have an email or need assistance logging <br /> in?
            <button onClick={() => {}}>GET HELP</button>{" "}
          </Forget>
          <br></br>
          <br></br>
          <Forget>
            <span onClick={() => {}}>LOG IN</span>{" "}
            <span onClick={() => {}}>{" . "}SIGN UP</span>{" "}
          </Forget>
        </form>
      </AuthContainer>
    </>
  );
};

export default ForgetPassword;
