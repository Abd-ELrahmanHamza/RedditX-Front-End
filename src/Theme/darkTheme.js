/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The secondary color of the website looks like primary in dark theme
    secondary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
    // The orange color of the website
    orange: "#d93a00",
    // The pink color of the website
    pink: "#fb133a",
    // The color of the text of the main links of search page
    titleText_of_link: "#D7DADC",

    // Some colors for authentication part
    logInButtonColorHovered: "#dc4a14",
    logInBlueHovered: "#3394dc",
    baseButtonColor: "#fff",
    subColor: "#797b7d",
    mainInputColor: "black",
    borderColor: "#86888a",
    blueValid: "#24a0ed",
    mainInputBackGroundColor: "#f6f7f8",
    initialBorderColor: "#e6e6e5",
    initialLabelColor: "#aeadad",
    errorColor: "#ea0027",
    orHeaderColor: "#93908e",
    authParagraphColor: "#1a1a1b",
    signInWithFacebookColor: "#237df3",
  },
  // The colors of the backgrounds in the website
  background: {
    // The primary color of the background on the website
    primary: "#1a1a1b",
    // The color of the muted background on the website
    muted: "#7c7c7c3d",
    // The pink color of the website
    pink: "#fb133a",
    // The light muted color of the website
    lightMuted: "#f6f7f8",
  },
  // Border colors
  borderColor: {
    // Primary border color
    primary: "#818384",
    // Secondary border color
    secondary: "#343536",
    // Muted border color
    muted: "#878a8c80",
    // The button hover color of the search page
    fill: "#2D2D2F",
    // The button active color of the search page
    active: "#272729",
    // button background color hover of the search page cards
    card_hover: "#353537",
    // Comment background color in search page
    comment: "#17232D",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
    // Post Border Line Color Search Page
    PostBorder: "#343536",
  },
  // The colors of the buttons in the website
  button: {
    // The color when hover on buttons on the website
    muted: "#1a1a1b",
  },
};

export default darkTheme;
