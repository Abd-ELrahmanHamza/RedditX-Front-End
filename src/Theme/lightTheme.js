/**
 * light theme colors
 *
 * @type {{id:string}}
 */
const lightTheme = {
  // The id of the theme
  id: "light",
  // The canvas color is the background color of the website
  canvas: "#DAE0E6",
};

export default lightTheme;
