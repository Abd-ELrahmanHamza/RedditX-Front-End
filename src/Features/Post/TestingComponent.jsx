// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
