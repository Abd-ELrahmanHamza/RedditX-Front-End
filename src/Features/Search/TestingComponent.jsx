// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
