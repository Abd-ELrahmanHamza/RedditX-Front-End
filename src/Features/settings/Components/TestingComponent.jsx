import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
