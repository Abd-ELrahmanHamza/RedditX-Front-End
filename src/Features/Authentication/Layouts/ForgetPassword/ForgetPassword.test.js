import { render, screen } from "@testing-library/react";
import TestingComponent from "Features/Post/TestingComponent";
import { BrowserRouter } from "react-router-dom";

import ForgetPassword from "./ForgetPassword";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
describe("ForgetPassword section", () => {
  it("renders ForgetPassword component", () => {
    render(
      <BrowserRouter>
        <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
          <ForgetPassword />
        </ThemeProvider>
      </BrowserRouter>
    );
  });
});
