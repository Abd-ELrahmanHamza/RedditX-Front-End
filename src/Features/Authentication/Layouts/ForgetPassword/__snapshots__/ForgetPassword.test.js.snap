// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgetPassword Component this is a test for PasswordStrength Component 1`] = `ShallowWrapper {}`;

exports[`ForgetPassword Layout this is a test for ForgetPassword Layout 1`] = `ShallowWrapper {}`;

exports[`ForgetPassword Layput this is a test for PasswordStrength Component 1`] = `ShallowWrapper {}`;

exports[`PasswordStrength Component this is a test for PasswordStrength Component 1`] = `ShallowWrapper {}`;
